pkg/operation/botanist/etcdencryption.go

```diff
@@ -178,7 +178,7 @@ func (b *Botanist) updateShootLabelsForEtcdEncryption(ctx context.Context, label
 }
 
 func (b *Botanist) persistEncryptionConfigInShootState(ctx context.Context) error {
-	gardenerResourceList := gardencorev1alpha1helper.GardenerResourceDataList(b.ShootState.Spec.Gardener)
+	gardenerResourceList := gardencorev1alpha1helper.GardenerResourceDataList(b.ShootState.Spec.Gardener).DeepCopy()
```

Comment:
What do you think about rather passing a `mutationFn` to `SaveGardenerResourcesInShootState` to spare the copy?

Reply:
Sounds good, I'll see if I can do it and maybe also adapt the code for the secret generations since I just saw that we have 3 copies going on there.